src/lib.rs

```diff
@@ -140,6 +142,23 @@ impl LndkOnionMessenger {
             return Err(());
         }
 
+        let version = client
+            .versioner()
+            .get_version(VersionRequest {})
+            .await
+            .expect("failed to get version")
```

Comment:
We're running a patched version of 0.17 (https://github.com/orbitalturtle/lnd/pull/14) which does support this. Is there any other way to check for compatibility that considers the node's supported features more directly in addition to or instead of looking at raw node version?

Reply:
Ah cool, I didn't realize that you were using that! Hmmmm I did a little digging but I can't think of another check that would quite work... What if we added a `--skip-version-check` flag option on start up that skips this check since you know the version you're running here is sufficient?

Reply:
I was thinking the feature bits from GetInfo could be checked? But if not, a `--skip-version-check` flag would work.

Reply:
If we can check it by the feature bits, I prefer that option.  Using `--skip-version-check` feels not ideal.
But we need 0.18 is because that 0.18 enables payments for blinded path, so only checking the feature bits seems not work.
Anyway, I'm new to this project so I'll follow your decision.

Reply:
Yeah, the thing is the custom LND version I made doesn't set that bit. But if you could track down that change in v18 and cherry-pick it to the custom version @mrfelton, we could probably get that working

Reply:
The onion messaging bit isn't set until LNDK starts running, so that won't work

Reply:
I'll add `--skip-version-check` option.

Reply:
You can skip version check with `--skip-version-check=true` option or adding `skip_version_check=true` to lndk.conf.